server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -1222,14 +1214,18 @@ default void onFailedEngine(String reason, @Nullable Exception e) {
         }
     }
 
-    public static final class Searcher implements Releasable {
+    public static final class Searcher extends IndexSearcher implements Releasable {
         private final String source;
-        private final IndexSearcher searcher;
         private final Closeable onClose;
 
-        public Searcher(String source, IndexSearcher searcher, Closeable onClose) {
+        public Searcher(String source, IndexReader reader,
```

Comment:
any chance we can make this `ElasticsearchDirectoryReader` instead of `IndexReader`?

Reply:
The `ElasticsearchDirectoryReader` can be wrapped by plugins so we'd need to move/recreate one when we create the [`Searcher`](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/index/shard/IndexShard.java#L1285). I agree that it would be simpler to always force the `ElasticsearchDirectoryReader` at the top though so I'll give it a try.

Reply:
I forgot the TranslogLeafReader that is used for get requests so this would require a bigger change and more refactoring. Maybe better to do this in a follow up ?

Reply:
sure ++ no rush I didn't anticipate this one either